### ISSUE TITLE
JSX element implicitly has type 'any' because no interface 'JSX.IntrinsicElements' exists. When using Preact v10.16.0

### ISSUE THREAD

**Issue by sant123** [denoland/fresh#1477]
![image](https://github.com/denoland/fresh/assets/7959437/6c8b65fa-f03b-46f2-9eba-88471b314850)

This is my deno.json

```json
{
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "preact"
  },
  "tasks": {
    "start": "deno run -A --watch=static/,routes/ dev.ts",
    "vpn": "DENO_DEPLOYMENT_ID=\"$(git rev-parse HEAD)\" deno run -A main.ts"
  },
  "imports": {
    "$fresh/": "https://deno.land/x/fresh@1.3.0/",
    "preact": "https://esm.sh/preact@10.16.0",
    "preact/": "https://esm.sh/preact@10.16.0/",
    "preact-render-to-string": "https://esm.sh/*preact-render-to-string@6.2.0",
    "@preact/signals": "https://esm.sh/*@preact/signals@1.1.5",
    "@preact/signals-core": "https://esm.sh/@preact/signals-core@1.3.1"
  }
}

```

Thanks!

**Comment by marvinhagemeister** [comment]
Can you check that the deno vscode plugin is initialized? There is a command for that in the vscode command palette.

**Comment by sant123** [comment]
Yep it is, in fact this is a project I'm currently running. Changing it to Preact v10.15.1 works good. That's my workaround for now.

**Comment by sant123** [comment]
[Screencast from 2023-07-18 14-03-35.webm](https://github.com/denoland/fresh/assets/7959437/b6593ab9-6a10-43b5-9577-78c1f509ac8c)

**Comment by afifurrohman-id** [comment]
> Screencast.from.2023-07-18.14-03-35.webm

**Solutions**
1. Ctrl/Cmd + Shift + P then choose **Reload Window** (vscode)
2. run fresh 
```sh
deno task start
```
and reload browser window (deno automatically cache missing dependencies)

**Comment by afifurrohman-id** [comment]
I use all latest preact / twind and it's work fine

**Comment by sant123** [comment]
Really weird @afifurrohman-id. I followed the steps you provided + I added --reload in my task to renew cache but still is showing that warning. 
![image](https://github.com/denoland/fresh/assets/7959437/7509e943-9ec5-4730-82d1-385b054fa8d5)

**Comment by sant123** [comment]
Alright I think I figured out the issue, but the workaround is a little odd to me:

To replicate the issue:

- Close any VSCode instance
- Delete Deno's cache. On Linux `rm -r $HOME/.cache/deno`
- Open a Fresh project with VSCode and select any .tsx file

You should see something like this:

![image](https://github.com/denoland/fresh/assets/7959437/ce70c25a-b622-4715-b046-e76cc4dbce07)

- Run `deno task start` 
- Now restart the Deno language server

![image](https://github.com/denoland/fresh/assets/7959437/583b15e3-cffd-462a-a1e8-e656c6fca943)

Now you should be able to see the error above:

![image](https://github.com/denoland/fresh/assets/7959437/225d6bb9-e1a0-480f-88ad-9a32cefec22f)

- To get rid of this annoying issue, stop the fresh server and run `deno check main.ts`

![image](https://github.com/denoland/fresh/assets/7959437/0d1d2a14-a5fc-4388-9a65-eed4f3c49c13)

![image](https://github.com/denoland/fresh/assets/7959437/c316c8e8-c634-4376-8bc1-6f90094d9a78)

My question is, why using `deno check` solves the issue? I tried running with both `dev.ts` and `main.ts` and did not work.

**Comment by mct-dev** [comment]
@sant123 this worked for me.  Idk why `deno check` works either, but it did.  I was just using the default fresh template from their "Getting Started", as well.  Might be worth fixing...

**Comment by marvinhagemeister** [comment]
FWIW For people commenting here: It's not an issue with Fresh but with Deno's LSP. We are aware of the issue but haven't found the root cause yet nor a reliable way to reproduce it. Sometimes I can reproduce it and when I try again it doesn't work anymore. My guess as to why `deno check` works is that it may refresh the internal type cache or something.

**Comment by marvinhagemeister** [comment]
I'll transfer this upstream to the deno cli repository, since this is not an issue with Fresh.
